.github/workflows/_rocm-test.yml

```diff
@@ -89,6 +83,25 @@ jobs:
         with:
           docker-image: ${{ steps.calculate-docker-image.outputs.docker-image }}
 
+      - name: Clean workspace
+        shell: bash
+        env:
+          DOCKER_IMAGE: ${{ inputs.docker-image }}
+        run: |
+          echo "${GITHUB_WORKSPACE}"
+          docker run --rm \
+            --user jenkins \
+            -v "${GITHUB_WORKSPACE}:/var/lib/jenkins/workspace" \
+            -w /var/lib/jenkins/workspace \
+            "${DOCKER_IMAGE}" \
+            sudo rm -rf /var/lib/jenkins/workspace/*
```

Comment:
@jeffdaily Cleaning up at the start of the workflow seems to be a better alternative to trying to clean up at the end of the workflow, which is what PR [123468](https://github.com/pytorch/pytorch/pull/123468) was trying to do (but getting into some complications trying to make sure the cleanup is triggered reliably for every workflow run). My only concern here is that instead of removing everything inside the workspace directory, what we really need is to change the ownership permissions for the workspace directory itself to undo the effects of https://github.com/pytorch/pytorch/pull/122922/files#diff-9709e5db13aeac90c0312b2b8da34b37cf51242ec789ca959fcf1ba295a8da7aR23?

Reply:
Removing workspace directory now.

Reply:
That's better, but still poses the question: if https://github.com/pytorch/pytorch/blob/666a628bea80a4f972a2ed4c01d6a0fe66508dd4/.github/actions/checkout-pytorch/action.yml#L31 does the same thing, isn't it better to just remove `no-sudo` for the `checkout-pytorch` step, since that's what your change is doing now anyway?

Reply:
Sorry, I realize my mistake: `checkout-pytorch` happens outside the docker container, so `sudo` won't work there.